File: backend/NotesApi/Models/Note.cs
using System.ComponentModel.DataAnnotations;

namespace NotesApi.Models;

public class Note
{
    public string UserId { get; set; } = string.Empty;
    public ApplicationUser User { get; set; } = null!;

    public int Id { get; set; }

    [Required]
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    // Navigation property - one note has many content items
    public List<ContentItem> ContentItems { get; set; } = new();

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/note.ts
export interface ContentItem {
  id: number;
  text: string;
  order: number;
  isStarred: boolean;
}

export interface Note {
  id: number;
  title: string;
  contentItems: ContentItem[];
  createdAt: string;
  updatedAt: string;
}

export interface CreateNote {
  title: string;
}

export interface UpdateNote {
  title: string;
  contentItems: string[];
}

export interface PatchContentItem {
  text?: string;
  isStarred?: boolean;
  order?: number;
}

export interface PatchNote {
  title?: string;
  contentItems?: string[];
}
